base/cvd/cuttlefish/host/commands/cvd/server_command/status_fetcher.h

```diff
@@ -18,38 +18,24 @@
 
 #include <sys/types.h>
 
-#include <string>
+#include <chrono>
 
 #include "common/libs/utils/result.h"
-#include "host/commands/cvd/command_request.h"
-#include "host/commands/cvd/instance_manager.h"
 #include "host/commands/cvd/selector/instance_group_record.h"
 #include "host/commands/cvd/selector/instance_record.h"
 
 namespace cuttlefish {
 
-struct StatusFetcherOutput {
-  std::string stderr_buf;
-  Json::Value json_from_stdout;
-  cvd::Response response;
-};
-
-class StatusFetcher {
- public:
-  StatusFetcher(InstanceManager& instance_manager)
-      : instance_manager_(instance_manager) {}
-  Result<StatusFetcherOutput> FetchStatus(const CommandRequest&);
-
-  Result<Json::Value> FetchGroupStatus(const CommandRequest& original_request,
-                                       selector::LocalInstanceGroup& group);
-
- private:
-  Result<std::string> GetBin(const std::string& host_artifacts_path) const;
-  Result<StatusFetcherOutput> FetchOneInstanceStatus(
-      const CommandRequest&, const InstanceManager::LocalInstanceGroup& group,
-      selector::LocalInstance&);
-
-  InstanceManager& instance_manager_;
-};
+// Fetches status from all instances in the group. Waits for the launcher to
+// respond for at most timeout seconds.
```

Comment:
nit: can the ` Waits for the launcher to respond for at most timeout seconds` part could be simplified as in `Fetches status from all instances in the group with a given timeout`?

Reply:
There is a difference between a timeout for the entire status command and a timeout for each of the communications with the launcher/run_cvd processes. Hopefully it's clearer now.